### PR TITLE
Introduce Add Component

### LINKED ISSUE #22: Add execution to Command class, parsing to Parser class
For each command listed in the [User Guide](https://ay2021s1-cs2113t-w13-3.github.io/tp/UserGuide.html), there needs to be:

* parsing done in the `Parser` class to prepare the arguments
* a corresponding `Command` child class that can execute the command with its parameters.

Commands given:

* `help`
* `template`
* `set`
* `add`
* `calc`

Parsing was already done in #9.

### PR DESCRIPTION
**Pull request by hughjazzman**
* Introduce `Add` component feature in #4 (Initial part of #5 as well).
* Refactor `Component` to use inheritance.
* Add `AddCommand` class to execute `add`.
* Update `Parser` to return `AddCommand`.
* Abstract `LoadComponent` for inheritance.
* Convert `Component` and `LoadComponent` to `abstract class`.

### CODE REVIEW

File: src/main/java/seedu/duke/component/Component.java [hunk 1]
```diff
@@ -2,8 +2,56 @@
...
+     * Returns the effective Component value when added in series.
+     *
+     * @param otherValue Component value.
+     * @return Effective value.
+     */
+    public double addSeries(double otherValue) {
+        if (isDirectSeries) {
+            return addDirect(otherValue);
+        } else {
+            return addReciprocal(otherValue);
```
Comment (sevenseasofbri) [inline src/main/java/seedu/duke/component/Component.java]: Perhaps the name otherValue is a little less intuitive? Consider naming it valueToBeAdded or something along those lines.

File: src/main/java/seedu/duke/commands/AddCommand.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        this.value = value;
+    }
+
+    /**
+     * Executes appropriate methods based on the given command.
+     *
+     * @param ui Ui object.
+     */
+    @Override
+    public void execute(Ui ui) throws DukeException {
```
Comment (sevenseasofbri) [inline src/main/java/seedu/duke/commands/AddCommand.java:27]: Unsure of the purpose of the ui object in this function

**Review by sevenseasofbri — COMMENTED** [review]
I think it mostly LGTM! Have provided some suggestions to change certain parts.

File: src/main/java/seedu/duke/commands/AddCommand.java (hunk 2 above)
Comment (hughjazzman) [reply src/main/java/seedu/duke/commands/AddCommand.java:27]: Will be used when it's actually implemented. The Ui will need to print what happens with execution, something like my iP. i.e. the implementation here is not complete, and requires #3 to be completed.

File: src/main/java/seedu/duke/component/Component.java (hunk 1 above)
Comment (hughjazzman) [reply src/main/java/seedu/duke/component/Component.java]: Hmm, I feel it's quite intuitive? What else would you do with the otherValue, with the method name and JavaDoc already given?

Comment (sevenseasofbri) [reply src/main/java/seedu/duke/component/Component.java]: I see. Okay it's alright then

File: src/main/java/seedu/duke/commands/AddCommand.java (hunk 2 above)
Comment (sevenseasofbri) [reply src/main/java/seedu/duke/commands/AddCommand.java:27]: Ohh okie alright gotcha

**Pull request merged**